src/settings/OrderNumber.js

```diff
@@ -22,44 +14,61 @@ class OrderNumber extends Component {
 
   constructor(props) {
     super(props);
+
     this.configManager = props.stripes.connect(ConfigManager);
   }
 
   getInitialValues = (settings) => {
-    const value = settings.length && settings[0].value === 'true';
+    const value = settings.length === 0 ? '' : settings[0].value;
```

Comment:
I'm not sure that I understand what is this for `settings[0].value === 'true'`... 
`const orderNumberSetting = get(settings, '0.value', '{}');`